autogen/retrieve_utils.py

```diff
@@ -297,3 +338,71 @@ def query_vector_db(
         where_document={"$contains": search_string} if search_string else None,  # optional filter
     )
     return results
+
+def create_lancedb_from_dir(dir_path, max_tokens, db_path, table_name, chunk_mode,
+                            must_break_at_empty_line, embedding_model_name):
+    """Create a LanceDB from all the files in a given directory."""
+    db = LanceDB.connect(db_path)
+    try:
+        # Load embedding model
+        embeddings = HuggingFaceEmbeddings(model_name=embedding_model_name, model_kwargs={'device': 'cpu'})
+
+        # Initialize your embedding function (replace with your actual embedding module)
+        embedding_function = SentenceTransformerEmbeddings(name=embedding_model_name)
+
+        table = db.get_table(table_name)
+
+        chunks = split_files_to_chunks(get_files_from_dir(dir_path), max_tokens, chunk_mode, must_break_at_empty_line)
+        print(f"Found {len(chunks)} chunks.")
+
+        for i, chunk in enumerate(chunks):
+            embeddings = embedding_function.generate_embeddings([chunk])  # Compute embeddings for the chunk
+            document = {
+                "vector": embeddings[0],  # Get the embedding for the single chunk
+                "text": chunk,
+                "id": f"doc_{i}",
+            }
+            table.insert(document)
+    except ValueError as e:
+        logger.warning(f"{e}")
+
+def query_lancedb(query_texts, n_results, db_path, table_name, search_string, embedding_model_name):
+    """Query a LanceDB."""
+    db = LanceDB.connect(db_path)
+    table = db.get_table(table_name)
+
+    # Initialize your embedding function (replace with your actual embedding module)
+    embedding_function = SentenceTransformerEmbeddings(name=embedding_model_name)
+
+    # Compute embeddings for the query texts
+    query_embeddings = embedding_function.generate_embeddings(query_texts)
+
+    # Query/search n most similar results
+    results = table.query(query_embeddings, n_results=n_results)
+
+    return results
+
+# Modify existing APIs to include the vector_database parameter
+def create_vector_db_from_dir(dir_path, max_tokens, client=None, db_path="/tmp/chromadb.db",
+                              collection_name="all-my-documents", get_or_create=False, chunk_mode="multi_lines",
+                              must_break_at_empty_line=True, embedding_model="all-MiniLM-L6-v2",
+                              vector_database="chromadb"):
```

Comment:
The `vector_database` should also be added to `retrieve_config` and the related code in `autogen/autogen/agentchat/contrib/retrieve_user_proxy_agent.py`

Reply:
where is `retrieve_config` im not getting

Reply:
It's defined in the file `retrieve_user_proxy_agent.py`.

Reply:
done,can you check